src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -336,8 +337,11 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                     $parameterData = $data[$key];
                     try {
                         if (null !== $constructorParameter->getClass()) {
+                            if (!$this->serializer instanceof DenormalizerInterface) {
+                                throw new LogicException(sprintf('Cannot create an instance of %s from serialized data because the injected serializer is not a denormalizer', $constructorParameter->getClass()));
```

Comment:
Similar exceptions above do not add such information (I guess the stacktrace is enough for this)